sentinel-core/src/main/java/com/alibaba/csp/sentinel/slots/block/flow/FlowRuleManager.java

```diff
@@ -129,21 +131,17 @@ public static boolean isOtherOrigin(String origin, String resourceName) {
         @Override
         public void configUpdate(List<FlowRule> value) {
             Map<String, List<FlowRule>> rules = FlowRuleUtil.buildFlowRuleMap(value);
-            if (rules != null) {
-                flowRules.clear();
-                flowRules.putAll(rules);
-            }
-            RecordLog.info("[FlowRuleManager] Flow rules received: {}", flowRules);
+            //the rules was always not null, it's no need to check nullable
+            //remove checking to avoid IDE warning
+            flowRules.set(rules);
+            RecordLog.info("[FlowRuleManager] Flow rules received: {}", flowRules.get());
```

Comment:
For eventual consistency maybe `rules` is better. Because there may be duplicated logs with same content when races occurred.

Reply:
Agree.